### PR TITLE
update lib

### PR DESCRIPTION
**Pull request by xwz123**

### CODE REVIEW

File: clear_labels.go [hunk 1]
```diff
@@ -4,21 +4,26 @@ import (
...
 
-	"github.com/opensourceways/community-robot-lib/giteeclient"
+	sdk "github.com/opensourceways/go-gitee/gitee"
 	"k8s.io/apimachinery/pkg/util/sets"
 )
 
-func (bot *robot) handleClearLabel(action string, prInfo giteeclient.PRInfo, cfg *botConfig) error {
-	if action != giteeclient.PRActionChangedSourceBranch {
+func (bot *robot) handleClearLabel(e *sdk.PullRequestEvent, cfg *botConfig) error {
+	action := sdk.GetPullRequestAction(e)
```
Comment (zengchen1024) [inline clear_labels.go:12]: don't code like C program

File: robot.go [hunk 2]
```diff
@@ -59,52 +58,51 @@ func (bot *robot) getConfig(cfg libconfig.PluginConfig, org, repo string) (*botC
...
+	if err = bot.handleClearLabel(e, cfg); err != nil {
 		merr.AddError(err)
 	}
 
-	commits := uint(e.PullRequest.Commits)
-	if err = bot.handleSquashLabel(action, prInfo, commits, cfg.SquashConfig); err != nil {
+	commits := uint(e.GetPullRequest().GetCommits())
+
+	err = bot.handleSquashLabel(e, commits, cfg.SquashConfig)
+	if err != nil {
```
Comment (zengchen1024) [inline robot.go:82]: if err = bot.handleSquashLabel(e, commits, cfg.SquashConfig); err != nil

File: squash.go [hunk 3]
```diff
@@ -1,19 +1,23 @@
...
+import sdk "github.com/opensourceways/go-gitee/gitee"
 
-func (bot *robot) handleSquashLabel(action string, prInfo giteeclient.PRInfo, commits uint, cfg SquashConfig) error {
+func (bot *robot) handleSquashLabel(e *sdk.PullRequestEvent, commits uint, cfg SquashConfig) error {
 	if cfg.unableCheckingSquash() {
 		return nil
 	}
 
-	if action != giteeclient.PRActionOpened && action != giteeclient.PRActionChangedSourceBranch {
+	action := sdk.GetPullRequestAction(e)
```
Comment (zengchen1024) [inline squash.go:10]: ditto

**Review by zengchen1024 — CHANGES_REQUESTED** [review]

File: robot.go [hunk 4]
```diff
@@ -59,52 +58,50 @@ func (bot *robot) getConfig(cfg libconfig.PluginConfig, org, repo string) (*botC
...
 
-	org, repo := ne.GetOrgRep()
+	org, repo := e.GetOrgRepo()
 	cfg, err := bot.getConfig(pc, org, repo)
 	if err != nil {
 		return err
 	}
 
-	toAdd, toRemove := getMatchedLabels(ne.GetComment())
+	toAdd, toRemove := getMatchedLabels(e.GetCommenter())
```
Comment (xwz123) [inline robot.go:100]: this is a bug: need use e.GetComment().GetBody()

File: clear_labels.go [hunk 5]
```diff
@@ -4,22 +4,26 @@ import (
...
```
Comment (xwz123) [inline clear_labels.go:12]: merge line 12 13

**Pull request merged**